### PR TITLE
SAK-31697 allow icons for tools that don't start with "sakai"

### PR DESCRIPTION
**Pull request by ern**

### CODE REVIEW

File: reference/library/src/morpheus-master/sass/base/_icons.scss [hunk 1]
```diff
@@ -6,10 +6,12 @@
...
 }
 
-
 @if $skin-with-icons {
-	[class^="icon-sakai"],
-	[class*=" icon-sakai"] {
+	.Mrphs-toolsNav__menuitem--icon,
+	.Mrphs-breadcrumb--icon,
+	ul li .selectedToolTitle i,
+	.removeTool,
```
Comment (buckett) [inline reference/library/src/morpheus-master/sass/base/_icons.scss]: This seems a rather generic selector `.removeTool`

File: reference/library/src/morpheus-master/sass/base/_icons.scss [hunk 2]
```diff
@@ -6,10 +6,12 @@
 	text-align: center;
 }
 
-
 @if $skin-with-icons {
-	[class^="icon-sakai"],
-	[class*=" icon-sakai"] {
```
Comment (payten) [inline reference/library/src/morpheus-master/sass/base/_icons.scss]: Hi @ern.  Is there a particular reason why we are removing the generic selectors?  These were added so we could easily include Sakai icons in any context without having to add to this selector.  I believe some tools that assume these generic selectors will now fail to show icons, including GradebookNG and Site Setup's Tool Order.

File: reference/library/src/morpheus-master/sass/base/_icons.scss (hunk 1 above)
Comment (ern) [inline reference/library/src/morpheus-master/sass/base/_icons.scss]: it seemed specific enough but can add some additional scope if needed.

**Comment by payten** [comment]
Sure thing.  I think I understand the problem and you're right, using `icon-` would conflict with the bootstrap icons.

The goal of the wildcard selectors was to allow the Sakai icon library (Font Awesome currently) to be used easily in any context in the application, without the developer having to define an explicit selector.  GradebookNG and Tool Order both use this to show tool icons, and doing a `git grep icon-sakai` I can see that Forums, Assignments and Site Info use the mechanism to show other Font Awesome icons.

Could an option 3) be we add those alternate prefixes to this list, e.g.
```
[class^="icon-sakai"],
[class*=" icon-sakai"],
[class^="icon-myprefix"],
[class*=" icon-myprefix"],
.icon-myawesometool,
.icon-myotherawesometool {
  @extend .fa-lg;
  @extend .fa;
}
```
This would be my preference, as it puts the onus on the tool developer to make one change rather than other UI developers having to add the selector of every spot they want to show a tool icon.

**Comment by payten** [comment]
Perhaps it's all much of a muchness... Do we want the onus on the developer adding the tool or the UI developer showing the icons? Maintenance of those specific selectors might become a headache, but maintaining a long list of icon prefixes is annoying too.  I'll leave all this up to the Morpheus folks to decide!

If we go with option 2), could you please add selectors for these cases:

Gradebook:
sak-31697__gradebookng

Tool Order
sak-31697__tool_order

Site Info tool:
sak-31697__site-info

Forums:
sak-31697__forums

Assignments:
sak-31697__assignments
sak-31697__assignments_2

**Comment by ern** [comment]
Hi Payten,

Thanks for all the thought on this...
The one thing that I liked the most was that someone could easily got to the file and with one line add their icon or update an existing. I would try to make it the easiest for the non-tech and simply want to add their icon or add a new one.

I would also appreciate others that would like to way in (nudge morpheus folks)

Maybe we just need a better namespace than "icon-", i.e. "sakai-icon--sakai-assignment"

**Comment by ern** [comment]
OK this PR has been completely revamped. The idea is to stick with the selector we had previously and create a namespace for icons in sakai:
**icon-sakai--**

Icons for tools use the prefix plus the tool id with any "**.**" converted to "**-**" in the tool id for example:
tool id -> css icon
**sakai.samigo => icon-sakai--sakai-samigo**

**Comment by ern** [comment]
@payten any comments on the new approach?

**Comment by payten** [comment]
Yep, looks good @ern.  It satisfies the "I can add a Sakai icon anywhere without having to change the CSS" case  - which is something I really like in the current implementation.

File: basiclti/basiclti-portlet/src/webapp/WEB-INF/sakai/IMSBLTIPortlet.xml [hunk 3]
```diff
@@ -95,7 +95,7 @@
 				<category name="portfolio" />
 
                 <configuration name="functions.require" />
-                <configuration name="sakai:menuClass" value="icon-sakai-lessonbuildertool"/>
+                <configuration name="sakai:menuClass" value="icon-sakai -2 dashes- sakai-lessonbuildertool"/>
```
Comment (fostersdesign) [inline basiclti/basiclti-portlet/src/webapp/WEB-INF/sakai/IMSBLTIPortlet.xml]: @ern, did you actually mean `-2 dashes-` here? Or do you mean `--`?

Comment (fostersdesign) [reply basiclti/basiclti-portlet/src/webapp/WEB-INF/sakai/IMSBLTIPortlet.xml]: Oh, I see why you did this: 

>  The string "--" is not permitted within comments.

Please ignore my previous comment.

**Pull request merged**